portfolio/src/main/java/com/google/sps/servlets/FormHandlerServlet.java

```diff
@@ -0,0 +1,33 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/form-handler")
+public class FormHandlerServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Get the value entered in the form.
+    String textValue = request.getParameter("text-input");
+
+    // Print the value so you can see it in the server logs.
+    System.out.println("You submitted the email address: " + textValue);
+
+    // Write the value to the response so the user can see it.
+    response.getWriter().println("You submitted the email address: " + textValue);
+
+    // Get the value entered in the form.
+    String message = request.getParameter("message-input");
+
+    // Print the value so you can see it in the server logs.
```

Comment:
I'm not sure if these comments are coming from the tutorial or not, but in general you want to keep comments to a minimum in the code. It sounds really counter intuitive, but the goal is that code should be readable on its own without comments to guide the reader. They're great for particular strange parts of code, commonly asked questions, and sort of "labeling" entire blocks of code for what they do, but otherwise, try to rely on them less.